src/Common/SensitiveDataMasker.cpp

```diff
@@ -74,19 +74,39 @@ class SensitiveDataMasker::MaskingRule
                 regexp_string_, regexp.error());
     }
 
-    uint64_t apply(std::string & data) const
+    uint64_t applyThrow(std::string & data) const
     {
         auto m = RE2::GlobalReplace(&data, regexp, replacement);
 
         if (throw_on_match && m > 0)
+        {
```

Comment:
btw it seems strange to have `throw_on_match = true/false` and `wipeSensitiveData(No)Throw` at the same time, e.g. now I'd expect we can remove `throw_on_match` and actually just use `wipeSensitiveData(No)Throw`  accordingly and a throw variant will always throw